### DISCUSSION TITLE
Adding card_image/thumbnail Support to Posts

### DISCUSSION THREAD

**Discussion by DuckerQA** [rochacbruno/marmite#118] [Ideas]
Hello, there! since I really like marmite I have an idea

To enhance post previews with visual appeal, consider adding support for a card_image or thumbnail field in the front matter of posts. This field would allow authors to specify an image that represents the content, which can be displayed in lists or previews.

Example in hello-word.md: 

```
---
title: "Example post"
date: 2024-10-25
tags: hello,world
card_image: ./media/duckerWizard.png. //new field
---

# Hello World

Hello, this is my first post on my new blog generated
by Marmite[1], a very simple to use blog generator.

```

Benefits:

- Improves the visual presentation of posts in listings.
- Provides a consistent way for authors to add and display post images.
- Enhances user engagement by making post previews more attractive.

**Comment by DuckerQA** [comment]
If the card_image functionality were available in the main context for both list and content templates, implementing the feature would be straightforward. The remaining adjustments to display thumbnails or images in list.html and content.html would be seamless and require minimal modifications. 

That's what it seems to me... 👍

**Reply by rochacbruno** [reply]
It is available.

On content

```
extra:
    card_image: url 
```

On template list HTML 

```
{{ content.extra.card_image }}
```

**Comment by rochacbruno** [comment]
this is planned on #12

**Comment by rochacbruno** [comment]
BTW: card_image is already a field.

And there is the `extra` that accepts anything you want for templates 


```yaml
---
extra:
  thumbnail: ./media/image.png
---

```

then on template 

{{ content.extra.thumbnail }}

**Comment by rochacbruno** [comment]
The current implementation supports extra: card_image 
https://github.com/rochacbruno/marmite/pull/112

That is used to generate the og:image meta tag and can also be
used to display on the list and post

**Comment by DuckerQA** [comment]
@rochacbruno I understand, huge thank you ! for your help and time!
